### ISSUE TITLE
Test case: mv_files_001_pos

### ISSUE THREAD

**Issue by behlendorf** [zfsonlinux/zfs#7242]
Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       | CentOS
Distribution Version    | 7
Linux Kernel                 | 4.8.1-1.el7.elrepo.x86_64
Architecture                 | x86_64
ZFS Version                  | zfs-0.7.0-334-g7088545
SPL Version                  | spl-0.7.0-29-g378c6ed

### Describe the problem you're observing

Rare failure of `mv_files_001_pos` test case.

### Describe how to reproduce the problem

Rarely reproduced by buildbot during automated testing.

### Include any warning/errors/backtraces from the system logs

http://build.zfsonlinux.org/builders/CentOS%207%20x86_64%20Mainline%20%28TEST%29/builds/5688

```
Test: /usr/share/zfs/zfs-tests/tests/functional/mv_files/mv_files_001_pos (run as root) [00:57] [FAIL]
07:42:01.73 ASSERTION: Doing a 'mv' of a large amount of files within a zfs filesystem works without errors.
07:42:58.95 The file number of target directory 3000 is not equal to that of the source  directory /mnt/testdir/newdir
```

**Comment by behlendorf** [comment]
Reopening.  Commit  d0f6fba did not fully address this issue.

**Comment by behlendorf** [comment]
Closing, resolved by 4f301661dfe7e90574ad634f8b7fe8da7e6b4064.
